docs/testing.md

```diff
@@ -0,0 +1,130 @@
+Once that design has been completed we can revise this document accordingly.
+
+### Common `10101-lightning-node` dependency between `app` and `coordinator`
+
+This `10101-lightning-node` is one of the aforementioned Rust crates shared between these two systems.
+It represents the custom Lightning Node built using (a fork of) `rust-lightning` and `rust-dlc`.
+
+### The `app`'s state will be controlled by its Rust backend
+
+Thanks to this we should be able to write system tests against the `app`'s API without involving the frontend.
+
+### The architecture will be similar to this component diagram
+
+This diagram does include some of the assumptions listed above:
+
+![Assumed component diagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/get10101/10101/d7619613324df3ce5c0f0a35d97bf24323f9515b/docs/diagrams/testing-component-diagram.puml)
+
+## Strategy
+
+In this section we describe the strategy that we recommend to follow when writing tests for this project.
+Each subsection corresponds to a section of the codebase or a combination thereof.
+
+### Common crates, `app` backend and `coordinator`
+
+All this code is written in Rust.
+
+#### Unit tests
+
+Good unit testing patterns apply.
+In particular, we strive for these tests to be small, structured[^1] and fast.
+Unit tests will ideally only be written for pure functions.
+
+They can be handy as API snapshot tests. For example, to ensure that a `serde` implementation doesn't change.
+
+#### Integration tests
+
+These are tests that depend on multiple components (perhaps across module boundaries).
+Additionally, this kind of tests usually involve mocked state.
+
+We are aiming to not spend too much time maintaining Rust integration tests, in favour of more comprehensive tests.
+
+### The `app` frontend
+
+This includes Flutter code (written in Dart) and the generated `flutter_rust_bridge`.
+In practice, there shouldn't be a strong need to write tests for the `flutter_rust_bridge` code we write (Rust) since it wil be a thin layer.
+
+#### Unit tests
+
+We can apply similar principles as the ones mentioned for Rust code.
+
+#### Widget tests
+
+Used to verify Flutter widget appearance and behaviour.
+They target a single widget.
+The state is mocked i.e. no Rust called via `flutter_rust_bridge`.
+
+#### Integration tests
+
+These are your traditional "UI tests".
+The clicks on the UI are simulated as if someone was using the application.
+
+They remain lightweight as they only call to a mocked version of the API provided by `flutter_rust_bridge`.
+We do not currently specify how this API shoud be mocked.
+We considered (1) [mocking the dependency in Dart](https://docs.flutter.dev/cookbook/testing/unit/mocking) and (2) mocking the dependency in Rust.
+We would prefer (1) as we think it would lead to simpler code than (2), but we are not sure if it's a feasible approach.
```

Comment:
I prefer (1) as well - since we not aim to test the generated flutter_rust_bridge code.